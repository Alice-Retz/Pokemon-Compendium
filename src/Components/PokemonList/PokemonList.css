.cardContainer {
	display: flex;
	flex-direction: row;
	flex-wrap: wrap;
	justify-content: center;
}
ul {
	list-style-type: none;
}
