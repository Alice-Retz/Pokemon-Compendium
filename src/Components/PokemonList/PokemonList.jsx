import React from 'react';
import PropTypes from 'prop-types';
import Pokemon from '../Pokemon/Pokemon';
import './PokemonList.css';

export default function PokemonList({ pokeArray }) {
	return (
		<ul className='cardContainer'>
			{pokeArray.map((pokeEntry) => {
				return (
					<li key={pokeEntry.pokeId}>
						<Pokemon pokeEntry={pokeEntry} />
					</li>
				);
			})}
		</ul>
	);
}

PokemonList.propTypes = {
	pokeArray: PropTypes.arrayOf(
		PropTypes.shape({
			pokeId: PropTypes.string.isRequired,
			name: PropTypes.string.isRequired,
			pic: PropTypes.string.isRequired,
			type1: PropTypes.string.isRequired,
			type2: PropTypes.string.isRequired,
			ability1: PropTypes.string.isRequired,
			ability2: PropTypes.string.isRequired,
			hiddenAbility: PropTypes.string.isRequired,
			eggGroup1: PropTypes.string.isRequired,
			eggGroup2: PropTypes.string.isRequired,
		}).isRequired
	),
};
