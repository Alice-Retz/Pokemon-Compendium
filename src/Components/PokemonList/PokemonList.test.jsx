import { render, screen } from '@testing-library/react';
import PokemonList from './PokemonList';

it('displays a list of pokemon', () => {
	render(
		<PokemonList
			pokeArray={[
				{
					pokeId: '5ff4fb7cd89993a89cc6544f',
					name: 'butterfree',
					pic: 'http://assets.pokemon.com/assets/cms2/img/pokedex/full/012.png',
					type1: 'bug',
					type2: 'flying',
					ability1: 'compound-eyes',
					ability2: 'NA',
					hiddenAbility: 'tinted-lens',
					eggGroup1: 'bug',
					eggGroup2: 'NA',
				},
				{
					pokeId: '5ff4fb7cd89993a89cc65444',
					name: 'venusaur-mega',
					pic: 'http://assets.pokemon.com/assets/cms2/img/pokedex/full/012.png',
					type1: 'grass',
					type2: 'poison',
					ability1: 'thick-fat',
					ability2: 'NA',
					hiddenAbility: 'NA',
					eggGroup1: 'grass',
					eggGroup2: 'poison',
				},
				{
					pokeId: '5ff4fb7cd89993a89cc65454',
					name: 'pidgeotto',
					pic: 'http://assets.pokemon.com/assets/cms2/img/pokedex/full/012.png',
					type1: 'normal',
					type2: 'flying',
					ability1: 'keen-eye',
					ability2: 'tangled-feet',
					hiddenAbility: 'big-pecks',
					eggGroup1: 'normal',
					eggGroup2: 'flying',
				},
			]}
		/>
	);

	const pokemons = screen.getByRole('list', { class: 'cardContainer' });
	expect(pokemons).toMatchSnapshot();
});
