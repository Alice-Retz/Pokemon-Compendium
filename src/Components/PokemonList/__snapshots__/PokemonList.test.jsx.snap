// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays a list of pokemon 1`] = `
<ul
  class="cardContainer"
>
  <li>
    <article
      class="pokeEntry"
    >
      <h2>
        butterfree
      </h2>
      <img
        alt="butterfree"
        src="http://assets.pokemon.com/assets/cms2/img/pokedex/full/012.png"
      />
      <div
        class="details"
      >
        <p>
          Type: 
          bug
          /flying
        </p>
        <p>
          Ability(s): 
          compound-eyes
          <br />
          Hidden Ability: 
        </p>
        <p>
          Egg Group(s): 
          bug
        </p>
      </div>
    </article>
  </li>
  <li>
    <article
      class="pokeEntry"
    >
      <h2>
        venusaur-mega
      </h2>
      <img
        alt="venusaur-mega"
        src="http://assets.pokemon.com/assets/cms2/img/pokedex/full/012.png"
      />
      <div
        class="details"
      >
        <p>
          Type: 
          grass
          /poison
        </p>
        <p>
          Ability(s): 
          thick-fat
          <br />
          Hidden Ability: 
        </p>
        <p>
          Egg Group(s): 
          grass
          /poison
        </p>
      </div>
    </article>
  </li>
  <li>
    <article
      class="pokeEntry"
    >
      <h2>
        pidgeotto
      </h2>
      <img
        alt="pidgeotto"
        src="http://assets.pokemon.com/assets/cms2/img/pokedex/full/012.png"
      />
      <div
        class="details"
      >
        <p>
          Type: 
          normal
          /flying
        </p>
        <p>
          Ability(s): 
          keen-eye
          /tangled-feet
          <br />
          Hidden Ability: 
        </p>
        <p>
          Egg Group(s): 
          normal
          /flying
        </p>
      </div>
    </article>
  </li>
</ul>
`;
